### PR TITLE
📊  war: change format of COW MID dataset

### PR DESCRIPTION
**Pull request by lucasrodes**
Simplify the COW MID dataset.

- Only report one metric: number of ongoing conflicts.
- Report disaggregated by `region` and `fatality`.
  - `fatality` describes a range of fatalities (lower and upper bounds) over the complete lifetime of a dispute globally.
  - We include an "unknown" category for `fatality`.

### CODE REVIEW

File: etl/steps/data/garden/war/2023-08-05/cow_mid.py [hunk 1]
```diff
@@ -125,39 +99,103 @@ def run(dest_dir: str) -> None:
...
+    tb.loc[tb["dispnum"] == 2328, "region"] = "Europe"
+    tb.loc[tb["dispnum"] == 4005, "region"] = "Asia"
+
+    # Check there is no NaN!
+    assert (tb.isna().sum() == 0).all(), "NaN in some field!"
+
+    return tb
 
 
 def add_regions(tb: Table) -> Table:
```
Comment (paarriagadap) [inline etl/steps/data/garden/war/2023-08-05/cow_mid.py:201]: The codes assigned are slightly different to what you defined on the yaml:
- Americas includes 1 here, but the yaml doesn't
- Middle East doesn't include 699 here, but the yaml does

**Review by paarriagadap — COMMENTED** [review]
Hi @lucasrodes! I added a few comments. I am pressing "Request changes" mainly because the grapher step for `cow_mid` is currently broken (though I know it's not in this PR, it's related with the changes):

```
(etl-py3.11) parriagadap@Pablos-MacBook-Pro etl % etl cow_mid
Detecting which steps need rebuilding...
OK (0.2s)                                                             
Running 2 steps:
1. data://garden/war/2023-08-05/cow_mid...
2023-08-18 11:15:12 [info     ] war.cow_mid: read and process MIDA table
2023-08-18 11:15:12 [info     ] war.cow_mid: read and process MIDB table
2023-08-18 11:15:12 [info     ] war.cow_mid: combine tables
2023-08-18 11:15:12 [info     ] war.cow_mid: estimate metrics
2023-08-18 11:15:12 [info     ] war.cow_mid: map fatality codes to names
2023-08-18 11:15:12 [info     ] war.cow_mid: replace NaNs with zeros where applicable
2023-08-18 11:15:12 [info     ] war.cow_mid: set index
2023-08-18 11:15:12 [info     ] war.cow_mid: add shortname to table
OK (1.6s)

2. data://grapher/war/2023-08-08/cow_mid...
2023-08-18 11:15:14,014 - [bugsnag] WARNING - No API key configured, couldn't notify
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/etl/run_python_step.py", line 34, in main
    _import_and_run(path, dest_dir)
  File "/Users/parriagadap/etl/etl/run_python_step.py", line 52, in _import_and_run
    module.run(dest_dir)
  File "/Users/parriagadap/etl/etl/steps/data/grapher/war/2023-08-08/cow_mid.py", line 33, in run
    tb = tb.set_index(["year", "country", "hostility_level"])
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/lib/catalog/owid/catalog/tables.py", line 532, in set_index
    t = super().set_index(keys, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/pandas/util/_decorators.py", line 331, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/pandas/core/frame.py", line 6009, in set_index
    raise KeyError(f"None of {missing} are in the columns")
KeyError: "None of ['hostility_level'] are in the columns"

COMMAND: poetry run run_python_step data://grapher/war/2023-08-08/cow_mid /Users/parriagadap/etl/data/grapher/war/2023-08-08/cow_mid
```

File: etl/steps/data/garden/war/2023-08-05/cow_mid.py [hunk 2]
```diff
@@ -125,39 +99,103 @@ def run(dest_dir: str) -> None:
...
+    """Combine MIDA and MIDB processed tables.
+
+    Basically, it adds region information (from MIDB) to MIDA.
+    """
+    # Merge
+    tb = tb_a.merge(tb_b, on=["dispnum", "year"], how="left")
+
+    # Fill NaNs
+    ## Sanity check (1)
+    dispnum_nans_expected = {2044, 2328, 4005}
```
[the remark this answers is not in the document]
Comment (lucasrodes) [reply etl/steps/data/garden/war/2023-08-05/cow_mid.py:182]: dispute codes; Ive clarified this with comments

**Review by paarriagadap — APPROVED** [review]

File: etl/steps/data/garden/war/2023-08-05/cow_mid.py (hunk 1 above)
Comment (lucasrodes) [reply etl/steps/data/garden/war/2023-08-05/cow_mid.py:201]: Aligning these, good catch!

**Comment by lucasrodes** [comment]
Thanks for reviewing, Pablo!

**Pull request merged**